### PR TITLE
fix: string field with false value in asconfig.js

### PR DESCRIPTION
**Pull request by woruyu**
Fixes https://github.com/AssemblyScript/assemblyscript/issues/2751

### CODE REVIEW

File: util/options.d.ts [hunk 1]
```diff
@@ -68,3 +68,6 @@ export function resolvePath(path: string, baseDir: string, useNodeResolution?: b
 
 /** Populates default values on a parsed options result. */
 export function addDefaults(config: Config, options: OptionSet): void;
+
+/** Sanitizes an option value to be a valid value of the option's type. */
+export function sanitizeValue(value: any, type: string): boolean | number | string | null | number[] | string[]; // eslint-disable-line @typescript-eslint/no-explicit-any
```
Comment (HerrCai0907) [inline util/options.d.ts:73]: unrelated change

Comment (CountBleck) [reply util/options.d.ts:73]: Hm, this is only necessary because this function is tested directly...is there a way to avoid that?

Comment (HerrCai0907) [reply util/options.d.ts:73]: Could you test this case by using `merge` API?

**Comment by HerrCai0907** [comment]
fixed in #2802
